_projects/enhance-and-improve-coala-community-and-projects-website.md

```diff
@@ -0,0 +1,160 @@
+organizations. The hurdle that newcomers mainly come across is the invitation
+to the organization before making any contributions, so this project adds a
+form for newcomers who wish to join the coala community. Also, to give a
+motivation to existing contributors, coala has created a `Promotion form` which
+will promote them to `coala Developer` status. To reduce the workload for
+maintainers, this project includes the development of form with some
+rudimentary checks that will promote the coala-newcomer to coala-developer.
+Moreover, this project adds major enhancements to websites that will help
+newcomers as well as developers to know more about this community and about
+what they have contributed to it.
+
+### Goals
+
+* Add calendar app to show timeline for various events
+* Add a one-stop tab for all google forms
+* Add contributors geolocation
+* Add *Login/SignUp* using GitHub
+* Add organization teams
+* Advanced filters
+* Automate welcoming process for newcomer
+* Automate newcomer promotion
+* Create new database models to manage data
+* Ease navigatibility in websites
+* Improve the **UI/UX of websites** to make more interactive
+* Motivating newcomers & developers with -
+  * modifications in Gamification Leaderboard
+  * adding a tab of **GSoC** in projects website
+* New coala-community logo
+* Contributors profile page
+* Admin/Maintainers dashboard
+
+### Milestones
+
+#### PREPARATION/BONDING
+
+* Detailed research on what **new cron jobs** or **API endpoints** need to be
+  added in [coala webservices](https://gitlab.com/coala/landing/).
+* Discussion on improvements to be made in coala [community website](http://community.coala.io/) and
+  [projects website](http://projects.coala.io/)
+* Research on new models to be created (if any) and adding new fields to
+  existing db models in coala community website **db models**.
+* *Created* and *Merged* a cEP of how the new enhancements will be implemented
+  and how will it be beneficial to [coala community](https://github.com/orgs/coala/teams/coala-newcomers).
+
+#### CODING PHASE 1
+
+1. Implementation of methods in **coala [webservices](https://gitlab.com/coala/landing)**
+2. Implementation of listed enhancements in coala [community website](http://community.coala.io/)
+
+    a. Create new **[Community Logo](https://github.com/coala/community/issues/55) + Header Image of README.md**
+
+    b. Add information about new endpoints at coala-webserivces homepage.
+
+    c. Add [contributor's Geolocation](https://github.com/coala/community/issues/18)
+
+    d. Add [Organisation Teams](https://github.com/coala/community/issues/12)
+
+    e. Add [Mentors](https://github.com/coala/community/issues/11)
+
+    f. Add **Newcomer Invitation** [form](https://github.com/coala/community/issues/89#issuecomment-468955279)
+
+    g. Add **Promotion to Developer** [form](https://github.com/coala/community/issues/89)
+
+    h. Manage live data of community members
```

Comment:
Could you explain a bit more here, what type of data you want to manage and how?

Reply:
A contributor gets assigned on issues, creates a PR for it and do reviews. Doing all this will eventually increase contribution time to the community with gain new skills as well as age (Earlier was eligible for GCI but now be eligible for GSoC)😄 
I want to manage the profile data with the issues (and PRs) the contributor has worked on. The `labels` of issues that will be displayed as a new column in gamification leaderboard. Also, the contributed time to the community. Also, looked over to fetch data that will be used to show statistics of a contributor in the form a bar/hike graph(2D or 3D). All this need to managed so that the website can help everyone.

Since managing all this data will require a number of API calls. So, I was thinking to call an `update function` in the background whenever contributor login and the data of that contributor gets updated with time.